src/main/java/teammates/ui/controller/InstructorStudentRecordsPageAction.java

```diff
@@ -114,5 +119,110 @@ private void filterFeedbackSessions(String courseId, List<FeedbackSessionAttribu
+     * @param giverEmails
+     * @return A map with instructor email => instructor name mappings.
+     */
+    private TreeMap<String, String> mapGiverNameToGiverEmail(String courseId, Set<String> giverEmails) {
+        TreeMap<String, String> giverEmailToGiverNameMap = new TreeMap<String, String>();
+        giverEmailToGiverNameMap.put(InstructorStudentRecordsPageData.COMMENT_GIVER_NAME_THAT_COMES_FIRST,
+                                     Const.DISPLAYED_NAME_FOR_SELF_IN_COMMENTS);
+        
+        // magic string from CommentsLogic
+        giverEmailToGiverNameMap.put("Anonymous", "Anonymous");
+        for (String giverEmail : giverEmails) {
+            if (!giverEmailToGiverNameMap.containsKey(giverEmail)) {
+                InstructorAttributes giverInstructor = logic.getInstructorForEmail(courseId, giverEmail);
+                Assumption.assertNotNull(giverInstructor);
+                giverEmailToGiverNameMap.put(giverEmail, giverInstructor.displayedName + " " + giverInstructor.name);
+            }
+        }
+        return giverEmailToGiverNameMap;
+    }
+    
+    /**
+     * Adds {@code comment} to {@code commentList} if the instructor is able to view the comment.
+     * @param comment
+     * @param isCurrentInstructorGiver
+     * @param commentList
+     * @param instructor
+     * @param courseId
+     */
+    private void updateCommentList(CommentAttributes comment, boolean isCurrentInstructorGiver,
+                                   List<CommentAttributes> commentList, InstructorAttributes instructor,
+                                   String courseId) {
+        boolean canViewComment = isCurrentInstructorGiver
+                                 || isInstructorAllowedForPrivilegeOnComment(comment, instructor, courseId,
+                                            Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_COMMENT_IN_SECTIONS);
+        if (canViewComment) {
+            commentList.add(comment);
+        }
+    }
+    
+    private boolean isInstructorAllowedForPrivilegeOnComment(CommentAttributes comment, InstructorAttributes instructor,
+                                                             String courseId, String privilegeName) {
+        
+        // student records only shows comments targeted at the student, and not team/section
+        if (instructor == null || comment.recipientType != CommentParticipantType.PERSON) {
+            return false;
+        }
+        
+        String studentEmail = "";
+        String section = "";
+        if (!comment.recipients.isEmpty()) {
+            Iterator<String> iterator = comment.recipients.iterator();
+            studentEmail = iterator.next();
+        }
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            return false;
+        } else {
+            section = student.section;
+            return instructor.isAllowedForPrivilege(section, privilegeName);
```

Comment:
It is ok to use `student.section` directly here. No need for the `section` variable.